django/views/generic/base.py

```diff
@@ -30,6 +32,27 @@ def get_context_data(self, **kwargs):
         return kwargs
 
 
+class ViewType(Enum):
```

Comment:
Can we use a `classproperty` on `View` instead? I'm not sure if we need `Enum` :thinking: e.g.
```python
class View:
    ...

    @classproperty
    def view_is_async(cls):
        handlers = [
            getattr(cls, method)
            for method in cls.http_method_names
            if (method != "options" and hasattr(cls, method))
        ]   
        if not handlers:
            return False
        is_async = asyncio.iscoroutinefunction(handlers[0])
        if not all(asyncio.iscoroutinefunction(h) == is_async for h in handlers[1:]):
            raise ImproperlyConfigured(
                f"{cls.__qualname__} HTTP handlers must either be all sync or all async."
            )   
        return is_async
```

Reply:
Yes... 🤔 I thought about this and preferred the enum. (Meh. I've left it for now, but can swap out if you feel strongly.)

Reply:
OK, I pushed the @classpropery version in 19f7a23e6e8b854c9ba2b83b3d8f1471307c0748. 

(Doing this, in `test_base_view_class_is_sync()` I initially made the mistake of comparing against the wrong Boolean value, which speaks to the value of the names enum values.)